### PR TITLE
Update comments for February 2021 LWG issues

### LINKED ISSUE #1679: February 2021 LWG issues
(Previous meta-issue: #1446)

At the February 2021 virtual plenary meeting, the following LWG issues were resolved in the C++ Working Paper.

:grey_question: Not yet analyzed
===
* Analyzed everything!

For each of these issues, we've determined whether they are:

:heavy_multiplication_x: Not applicable
===
If an issue requires no action from implementers, we mark it as N/A. Categories:

* Pure wording clarifications with nothing to implement (these can be changes to non-normative text like examples and informative notes, or wording cleanups to normative text that don't impact observable behavior)
  + [x] LWG-3495 `constexpr launder` makes pointers to inactive members of unions usable
* Something that increases the restrictions placed on users, but implementers aren't expected to enforce those restrictions
* Fixes for obviously broken wording, where implementers would have done the right thing anyways

:smile_cat: Already implemented
===
Sometimes we cite LWG issues in product code comments as we're implementing their proposed resolutions. When the resolutions are officially accepted, we should remove the citations (as the default assumption is that we're implementing what the Standard says). If something is especially subtle, we can convert the citation to mention the relevant Standard section.

Sometimes we should add test coverage - e.g. when the Standard begins requiring something that we were already doing, but weren't explicitly testing for.

* Already implemented, comments need to be removed and messages need to cite the Standard - created PR #1686:
  + [x] LWG-3391 Problems with `counted_iterator/move_iterator::base() const &`
  + [x] LWG-3433 `subrange::advance(n)` has UB when `n < 0`
    - Cited as LWG-3413 due to a typo
  + [x] LWG-3490 `ranges::drop_while_view::begin()` is missing a precondition
  + [x] LWG-3492 Minimal improvements to `elements_view::`*`iterator`*
* Already implemented without comments

:adhesive_bandage: Patches an unimplemented feature
===
We should record this LWG issue in the GitHub issue tracking the feature. That way, we'll remember to verify it, but it doesn't represent net new work.

* Recorded in issue #39 and PR #1436:
  + [x] LWG-3500 `join_view::iterator::operator->()` is bogus
* Recorded in issue #39:
  + [x] LWG-3505 `split_view::`*`outer-iterator`*`::operator++` misspecified

:beetle: Not yet implemented
===
We should file a [GitHub issue labeled **LWG**](https://github.com/microsoft/STL/issues?q=is%3Aissue+label%3ALWG); see those existing examples.

* [x] LWG-3494 Allow ranges to be conditionally borrowed
  + Filed issue #1682.
* [x] LWG-3502 `elements_view` should not be allowed to return dangling references
  + Filed issue #1685.

### PR DESCRIPTION
**Pull request by StephanTLavavej**
Fixes #1679.

Adds a WP citation to `<strstream>` for consistency (the same WP is being cited repeatedly, but the occurrences are separated by a few lines).

### CODE REVIEW

File: stl/inc/xutility
Comment (CaseyCarter) [inline stl/inc/xutility:3576]: Ditto missing document number.

This comment was here to explain why the behavior differed from the spec (poorly, given that the issue number was wrong). It now seems strange to have a comment that says "this behaves as the spec says it should". Strike it?

File: stl/inc/ranges
Comment (CaseyCarter) [inline stl/inc/ranges:2199]: This citation needs a document number. (This will be hard to obtain until there's a new working draft.)

File: tests/std/tests/P0896R4_views_drop_while_death/test.cpp
Comment (CaseyCarter) [inline tests/std/tests/P0896R4_views_drop_while_death/test.cpp:24]: Ditto "missing document number".

**Review by CaseyCarter — CHANGES_REQUESTED** [review]

File: stl/inc/xutility
Comment (StephanTLavavej) [reply stl/inc/xutility:3576]: Yeah, we can drop this comment (I wasn't sure if you still wanted it, so I erred on the side of retaining it).

File: stl/inc/ranges
Comment (StephanTLavavej) [reply stl/inc/ranges:2199]: We can wait for the new WP. Marking as blocked until then.

Comment (CaseyCarter) [reply stl/inc/ranges:2199]: The next Working Draft will be N4885 (https://github.com/cplusplus/draft/commit/bd2cc23e7cc72d5fc0445da1955dc4b46f7caa25. I'll unblock this PR, confident that we can come back and fix things up in the unlikely case that WG21 repeats the "same document number assigned to two different documents" fiasco.

File: stl/inc/strstream
Comment (CaseyCarter) [inline stl/inc/strstream:254]: This is para 16 in N4885 if you'd like to avoid referencing the "ancient" N4727. I'm not bothered enough to Require Changes.

**Review by CaseyCarter — APPROVED** [review]

File: stl/inc/strstream
Comment (StephanTLavavej) [reply stl/inc/strstream:254]: Thanks - multiple lines are affected, so I don't want to extend this PR more. We have #182 tracking this.

**Review by MahmoudGSaleh — APPROVED** [review]

**Pull request merged**
